multinode-demo/common.sh

```diff
@@ -188,10 +188,10 @@ oom_score_adj() {
   fi
 }
 
-SOLANA_CONFIG_DIR=${SNAP_DATA:-$PWD}/config
-SOLANA_CONFIG_PRIVATE_DIR=${SNAP_DATA:-$PWD}/config-private
-SOLANA_CONFIG_VALIDATOR_DIR=${SNAP_DATA:-$PWD}/config-validator
-SOLANA_CONFIG_CLIENT_DIR=${SNAP_USER_DATA:-$PWD}/config-client
+SOLANA_CONFIG_DIR=${SNAP_DATA:-~/.solana}/config
```

Comment:
I'm not a fan of writing outside of the tree.  But if we do it then I think we need to generate a unique dir, perhaps with `mktemp`?  This as written will cause a collision when using multiple workspaces on the same machine

Reply:
I can key the ~/.solana entry to a hash of the working tree to support multiple.  Fact is these scripts don't support multiple working trees as-is with the networking conflicts...

mktemp would require we somehow carry that info across the various scripts, which are already conveying too much.

Reply:
What's motivating ~/.solana?  This is essentially creating a global variable without any kind of synchronization and a memory leak.

Reply:
cargo build times :-(

Reply:
Yeah I was hoping we’d first get to the bottom of why this is even an issue first

Reply:
I'd be ok with landing ~/.solana with the promise that we revert it as soon as we figure out why incremental `cargo build`s are busted

Reply:
And a promise comes in the form of an assigned github issue attached to a milestone before this PR is merged? :)